src/bin/rugs_metadata_server.rs

```diff
@@ -216,6 +225,44 @@ mod tests {
         Ok(pool)
     }
 
+    /// Test reading auth config secrets from files
+    #[tokio::test]
+    async fn config_secrets() -> Result<()> {
```

Comment:
This is great, thank you! It would be awesome if you also added a test for the old environment variables, to make sure there's no regression in parsing from those, and a test for what happens if both are specified. (I don't particularly care which outcome we pick -- whether we prefer the file option or the env var option -- but having a test means we'll have a failure if the behavior ever changes, and people might unknowingly be depending on whatever behavior we introcuce.)